src/ansible_compat/runtime.py

```diff
@@ -423,11 +425,18 @@ def install_collection(
         if force:
             cmd.append("--force")
 
+        if isinstance(collection, Path):
+            collection = str(collection)
         # As ansible-galaxy install is not able to automatically determine
         # if the range requires a pre-release, we need to manuall add the --pre
         # flag when needed.
-        matches = version_re.search(str(collection))
-        if matches and CollectionVersion(matches[1]).is_prerelease:
+        matches = version_re.search(collection)
+
+        if (
+            not str(collection).startswith("git+")
```

Comment:
fwiw, the core logic also includes `git@`: https://github.com/ansible/ansible/blob/3e9d374fda1c91af6cefc88cf7d12916dc98f00b/lib/ansible/galaxy/dependency_resolution/dataclasses.py#L151-L152

Reply:
Thanks. Please have another look, I made some changes to allow both.